src/backend/distributed/transaction/backend_data.c

```diff
@@ -862,12 +862,11 @@ CancelTransactionDueToDeadlock(PGPROC *proc)
  * transaction was cancelled due to a deadlock. If the backend is not in a
  * distributed transaction, the function returns false.
  * We keep some session level state to keep track of if we were cancelled
- * because of a distributed deadlock. When clearState is true, this function
- * also resets that state. So after calling this function with clearState true,
- * a second would always return false.
+ * because of a distributed deadlock. This function also resets that state.
+ * So a second call made to this function would always return false.
  */
 bool
-MyBackendGotCancelledDueToDeadlock(bool clearState)
+MyBackendGotCancelledDueToDeadlock(void)
```

Comment:
this is intentional, because the function has a side-effect. We should signal that to the readers somehow. So, I'd rather keep it, as a reader you know that you are changing the state when you pass a flag, otherwise we should rename: `GetAndClearMyBackendGotCancelledDueToDeadlock` which might be fine

Reply:
Instead of a longer function name, I guess I'm more inclined to keep the function as is, thank you for telling me the reason behind :)